src/main/kotlin/fm/unit/dao/Reports.kt

```diff
@@ -14,11 +18,46 @@ data class Report(
         val prefix: String)
 
 interface Reports {
+
+    @CreateSqlObject
+    fun testsuites(): Testsuites
+
+    /**
+     * Insert a new report.
+     *
+     * Note that the report is empty unless testsuites are inserted that belong to this report.
+     *
+     * @param organization_id The organization of this report.
+     * @param repository_id The repository this report belongs to.
+     * @param commit_hash The commit the report belongs to.
+     * @param prefix A common prefix to group reports, e.g. `system-test`.
+     * @return The id of the inserted report.
+     */
     @SqlUpdate("""
         INSERT INTO reports (organization_id, repository_id, commit_hash, prefix)
         VALUES (:organization_id, :repository_id, :commit_hash, :prefix)
     """)
-    fun insert(@BindBean report: Report)
+    @GetGeneratedKeys
+    fun insert(organization_id: Int, repository_id: Int, commit_hash: String, prefix: String): Int
+
+    /**
+     * Insert a new report with its testsuite, ie JUnit XML files.
+     *
+     * @param organization_id The organization of this report.
+     * @param repository_id The repository this report belongs to.
+     * @param commit_hash The commit the report belongs to.
+     * @param prefix A common prefix to group reports, e.g. `system-test`.
+     * @param suites A list of the testsuites that will e inserted.
+     * @return The id of the inserted report.
+     */
+    @Transaction
+    fun insert(organization_id: Int, repository_id: Int, commit_hash: String, prefix: String, suites: List<Testsuite>): Int {
+        val suite_dao = testsuites()
```

Comment:
nit: `s/suite_dao/suiteDao | suites`? But it's ok for `report_id` since those correlate to table fields